app/hooks/use-interval.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useRef } from 'react';
+
+export function useInterval(callback: () => void, delay: number) {
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  const savedCallback = useRef(() => {});
+
+  useEffect(() => {
+    savedCallback.current = callback;
+  });
```

Comment:
I'll be honest that this was a copy paste job. I believe it's needed as the ref is persisted during rerenders of the component rather than being reinitialised, though I'm not 100%

Reply:
You can use `useRef` as you would use `useCallback` to ensure there is only one version of the function, rather than as Kyran said, it being reinitialized on each render.